src/mesh/Vertex.hpp

```diff
@@ -53,6 +53,9 @@ class Vertex {
 
   void tag();
 
+  /// Returns a coordinate of a vertex
+  inline double coord(int index) const;
```

Comment:
Do we want to allow for negatives? Would an `std::size_t` fit better?
Also in further places (since these are new methods).
It probably does not matter at all in this context.

I guess a strong type (e.g., `CoordID`) would just be painful without reason.

Reply:
This should actually be VertexID instead of bare int.

This is a pretty general issue we have in precice. VertexID is an int, Eigen uses int by default, stl styles use size_t and we normally use int for loops.

Moving everything to size_t may be a solution, but I am afraid that would be a v4 topic.

Reply:
This is essentially `0`, `1`, or `2` for 3D.

`std::size_t` is a bit overkill for something that small. `unsigned char` would probably be the right choice here. Let's solve this in another PR though.

Reply:
Opened an issue https://github.com/precice/precice/issues/1970